### PR TITLE
[kodi] Added NPE safe-guard and error handling

### LINKED ISSUE #2608: [kodi] NPE after receiving "OnPlay" event from server
## Current behavior

log in openHAB 2.1.0 -  snapshot 1020 -
```
2017-08-28 22:11:28.089 [DEBUG] [i.internal.protocol.KodiClientSocket] - Message received from server: {"jsonrpc":"2.0","method":"Player.OnPlay","params":{"data":{"item":{"title":"Dagsrevyen 21","type":"movie"},"player":{"playerid":-1,"speed":1}},"sender":"xbmc"}}
2017-08-28 22:11:28.089 [DEBUG] [i.internal.protocol.KodiClientSocket] - Event received from server: {"jsonrpc":"2.0","method":"Player.OnPlay","params":{"data":{"item":{"title":"Dagsrevyen 21","type":"movie"},"player":{"playerid":-1,"speed":1}},"sender":"xbmc"}}
2017-08-28 22:11:28.099 [DEBUG] [i.internal.protocol.KodiClientSocket] - send message: {"jsonrpc":"2.0","id":3009,"method":"Player.GetItem","params":{"playerid":-1,"properties":["title","album","artist","director","thumbnail","file","fanart","showtitle","streamdetails","channel","channeltype"]}}
2017-08-28 22:11:28.100 [DEBUG] [i.internal.protocol.KodiClientSocket] - Message received from server: {"error":{"code":-32602,"data":{"method":"Player.GetItem","stack":{"message":"Value between 0 (inclusive) and 2 (inclusive) expected but -1 received","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":3009,"jsonrpc":"2.0"}
2017-08-28 22:11:28.101 [DEBUG] [i.internal.protocol.KodiClientSocket] - Response received from server: {"error":{"code":-32602,"data":{"method":"Player.GetItem","stack":{"message":"Value between 0 (inclusive) and 2 (inclusive) expected but -1 received","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":3009,"jsonrpc":"2.0"}
2017-08-28 22:11:28.102 [DEBUG] [i.internal.protocol.KodiClientSocket] - callMethod returns {"error":{"code":-32602,"data":{"method":"Player.GetItem","stack":{"message":"Value between 0 (inclusive) and 2 (inclusive) expected but -1 received","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":3009,"jsonrpc":"2.0"}
2017-08-28 22:11:28.104 [INFO ] [.eclipse.smarthome.model.script.Kodi] - KodiStopped.state=OFF
2017-08-28 22:11:28.106 [ERROR] [i.internal.protocol.KodiClientSocket] - Error handling event {"jsonrpc":"2.0","method":"Player.OnPlay","params":{"data":{"item":{"title":"Dagsrevyen 21","type":"movie"},"player":{"playerid":-1,"speed":1}},"sender":"xbmc"}} player state change message: null
java.lang.NullPointerException: null
        at org.openhab.binding.kodi.internal.protocol.KodiConnection.requestPlayerUpdate(KodiConnection.java:234) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at org.openhab.binding.kodi.internal.protocol.KodiConnection.processPlayerStateChanged(KodiConnection.java:377) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at org.openhab.binding.kodi.internal.protocol.KodiConnection.handleEvent(KodiConnection.java:354) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at org.openhab.binding.kodi.internal.protocol.KodiClientSocket$KodiWebSocketListener$2.run(KodiClientSocket.java:158) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

## Related sources
Related discussion in community: https://community.openhab.org/t/kodi-binding-exception-when-starting-pvr-recording/33674

### PR DESCRIPTION
**Pull request by cweitkamp**
Closes #2608 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java [hunk 1]
```diff
@@ -222,14 +222,20 @@ public synchronized JsonElement callMethod(String methodName, JsonObject params)
...
             sendMessage(message);
             if (commandLatch.await(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
-                logger.debug("callMethod returns {}", commandResponse.toString());
-                return commandResponse.get("result");
+                logger.debug("callMethod returns {}", commandResponse);
+                if (commandResponse.has("result")) {
+                    return commandResponse.get("result");
+                } else {
+                    JsonElement error = commandResponse.get("error");
+                    logger.debug("Error received from server: {}", error != null ? error : "");
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java]: Why not: 
`logger.debug("Error received from server: {}", error);`

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java [hunk 2]
```diff
@@ -222,14 +222,20 @@ public synchronized JsonElement callMethod(String methodName, JsonObject params)
...
+                    return null;
+                }
             } else {
-                logger.error("Timeout during callMethod({}, {})", methodName, params != null ? params.toString() : "");
+                logger.debug("Timeout during callMethod({}, {})", methodName, params != null ? params : "");
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java]: Maybe also change to 
`logger.debug("Timeout during callMethod({}, {})", methodName, params);`

**Comment by cweitkamp** [comment]
@martinvw Thanks for review. What a pitty. Why didn't I realized those my own?!

**Comment by martinvw** [comment]
Thanks

**Pull request merged**